### ISSUE TITLE
eslint reports max-depth for `else if` blocks.

### ISSUE THREAD

**Issue by Raynos** [eslint/eslint#1835]
When specifying `max-depth` I expected it to not complain about nested else if statements.

I would expect that max-depth would only complain when it's indented 4 times.

```js
/*global require*/
/*eslint global-strict: [2, "always"], quotes: [2, "single"]*/

/*eslint max-depth: [2, 4]*/
'use strict';

var process = require('process');

function log(m) {
    process.stdout.log(m + '\n');
}

function foo(a) {
    if (a === 1) {
        log('1');
    } else if (a === 2) {
        log('2');
    } else if (a === 3) {
        log('3');
    } else if (a === 4) {
        log('4');
    } else if (a === 5) {
        log('5');
    } else if (a === 6) {
        log('6');
    }
}

var x = 3;
foo(x);

/*
    $ eslint test.js

    test.js
      21:11  error  Blocks are nested too deeply (5)  max-depth
      23:11  error  Blocks are nested too deeply (6)  max-depth

        ✖ 2 problems (2 errors, 0 warnings)
*/
```

Sounds like the fix would be to decrement the max depth counter on an `else if` statement.

cc @malandrew @jcorbin

**Comment by nzakas** [comment]
Definitely looks like a bug.

**Comment by nzakas** [comment]
Looking at this now.
